src/arch/x86/reg/amd64_core.rs

```diff
@@ -0,0 +1,155 @@
+use crate::arch::Registers;
+use core::convert::TryInto;
+
+/// 64-bit x86 core registers.
+///
+/// Source: https://github.com/bminor/binutils-gdb/blob/master/gdb/features/arm/arm-core.xml
+#[derive(Default)]
+pub struct Amd64CoreRegs {
+    // RAX, RBX, RCX, RDX, RSI, RDI, RBP, RSP, r8-r15
+    regs: [u64; 16],
+    // Status register
+    eflags: u32,
+    // Instruction pointer
+    rip: u64,
+    // Segment registers: CS, SS, DS, ES, FS, GS
+    segments: [u32; 6],
+    // FPU registers: ST0 through ST7
+    st_regs: [F80; 8],
+    // SIMD Registers: XMM0 through XMM15
+    xmm_regs: [u128; 0x10],
```

Comment:
hmm, `xmm` and `mxcsr` aren't technically part of `i386/64bit-core.xml`, and seem to be defined in `i386/64bit-sse.xml`.

That said, I kinda fudged things myself when implementing ARM, and included the FPU registers (defined in `arm-fpa.xml`) in the "core" registers, so I guess we can just fudge it for x86 as well...